src/editor/EditorManager.js

```diff
@@ -779,6 +778,38 @@ define(function (require, exports, module) {
         }
     }
 
+    /**
+    * @param {Object} editor - function that is called to validate provider
+    * @return {String} id - name of the provider so we can return it to know which
+    * provider we have found
+    */
+    function findProvider(editor) {
+        var i;
+
+        for(i = 0 ; i < _inlineproviders.length; i++){
+            var provider = _inlineproviders[i].provider;
+            if(provider(editor)){
+                return _inlineproviders[i].id;
```

Comment:
It seems odd to me that you're returning an `id` vs. the provider itself.  Why that choice?